File: C#/KthSmallest.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    
    public List<int> TreeValues = new List<int>();
    
    public int KthSmallest(TreeNode root, int k) {
        
        TraverseTree(root);
        
        for (int i = 0; i < k - 1; i++)
        {
            TreeValues.Remove(TreeValues.Min());
        }
        
        return TreeValues.Min();
    }
    
    public void TraverseTree(TreeNode root)
    {
        TreeValues.Add(root.val);
        //Console.WriteLine("Node " + root.val);
        
        if (root.left != null)
        {
            TraverseTree(root.left);
        }
        
        if (root.right != null)
        {
            TraverseTree(root.right);
        }
        
        return;
    }
}

File: Python/kthSmallest.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:

        rootValues = []

        def findKSmall(root):

            nonlocal rootValues

            if not(root):
                return

            findKSmall(root.left)
            findKSmall(root.right)

            rootValues.append(root.val)

        findKSmall(root)

        rootValues.sort()

        # print(rootValues)

        return rootValues[k - 1]